opsimulator/opsimulator_test.go

```diff
@@ -0,0 +1,243 @@
+package opsimulator
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-service/predeploys"
+	"github.com/ethereum-optimism/supersim/bindings"
+	"github.com/ethereum-optimism/supersim/config"
+	"github.com/ethereum-optimism/supersim/testutils"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/stretchr/testify/require"
+)
+
+// MockChainWithDependencySet extends MockChain to support dependency set configuration
+type MockChainWithDependencySet struct {
+	*testutils.MockChain
+	chainID       uint64
+	dependencySet []uint64
+}
+
+func NewMockChainWithDependencySet(chainID uint64, dependencySet []uint64) *MockChainWithDependencySet {
+	return &MockChainWithDependencySet{
+		MockChain:     testutils.NewMockChain(),
+		chainID:       chainID,
+		dependencySet: dependencySet,
+	}
+}
+
+func (c *MockChainWithDependencySet) Config() *config.ChainConfig {
+	return &config.ChainConfig{
+		Name:      "mockchain",
+		ChainID:   c.chainID,
+		BlockTime: 2,
+		L2Config: &config.L2Config{
+			L1ChainID:     900,
+			DependencySet: c.dependencySet,
+		},
+	}
+}
+
+// createMockExecutingMessage creates a mock CrossL2InboxExecutingMessage for testing
+func createMockExecutingMessage(sourceChainID uint64) *bindings.CrossL2InboxExecutingMessage {
+	return &bindings.CrossL2InboxExecutingMessage{
+		MsgHash: [32]byte{0x12, 0x34}, // mock message hash
+		Id: bindings.Identifier{
+			Origin:      common.HexToAddress("0x1234567890123456789012345678901234567890"),
+			BlockNumber: big.NewInt(100),
+			LogIndex:    big.NewInt(0),
+			Timestamp:   big.NewInt(1234567890),
+			ChainId:     big.NewInt(int64(sourceChainID)),
+		},
+		Raw: types.Log{
+			Address:     predeploys.CrossL2InboxAddr,
+			Topics:      []common.Hash{crypto.Keccak256Hash([]byte("ExecutingMessage"))},
+			Data:        []byte("mock data"),
+			BlockNumber: 100,
+		},
+	}
+}
+
+func TestCheckInteropInvariants_DependencySetValidation(t *testing.T) {
```

Comment:
would be good to add a test case that validates that a message with same source and destination, passes validation